openscenario/openscenario_visualization/src/openscenario_visualization_condition_groups_plugin/openscenario_visualization_condition_groups_plugin.cpp

```diff
@@ -0,0 +1,273 @@
+  for (const auto & condition_groups : *condition_groups_collection_ptr_) {
+    context_ss << std::fixed << std::setprecision(0)
+               << "Condition Groups Name: " << condition_groups.groups_name << std::endl;
+    for (const auto & condition_group : condition_groups.condition_groups) {
+      for (const auto & condition : condition_group.conditions) {
+        context_ss << "  Current Evaluation: " << condition.current_evaluation << std::endl
+                   << "  Current Value: " << condition.current_value << std::endl
+                   << "  Type: " << condition.type << std::endl;
+      }
+    }
+    context_ss << std::endl;
+  }
+
+  painter.drawText(
+    property_left_->getInt(), property_top_->getInt(), overlay_->getTextureWidth(),
+    overlay_->getTextureHeight(), Qt::AlignLeft | Qt::AlignTop, context_ss.str().c_str());
+
+  painter.end();
+  last_msg_ptr_ = msg_ptr;
+}
+
+void VisualizationConditionGroupsDisplay::updateVisualization()
+{
+  int width = property_width_->getInt();
+  int height = property_length_->getInt();
+
+  overlay_->updateTextureSize(width, height);
+  overlay_->setPosition(property_left_->getInt(), property_top_->getInt());
+  overlay_->setDimensions(width, height);
+
+  if (last_msg_ptr_) {
+    processMessage(last_msg_ptr_);
+  }
+}
+
+void VisualizationConditionGroupsDisplay::loadConditionGroups(const Context::ConstSharedPtr msg_ptr)
+{
+  if (!msg_ptr) return;
+
+  YAML::Node data;
+  try {
+    data = YAML::Load(msg_ptr->data);
+  } catch (const std::exception & e) {
+    throw std::runtime_error(std::string("Failed to load YAML: ") + e.what());
+  }
+
+  int unnamed_event_counter = 1;
+  condition_groups_collection_ptr_ = std::make_shared<ConditionGroupsCollection>();
+
+  auto stories = data["OpenSCENARIO"]["Storyboard"]["Story"];
+  for (const auto & story : stories) {
+    processStory(story, unnamed_event_counter);
+  }
+}
+
+void VisualizationConditionGroupsDisplay::processStory(const YAML::Node & story_node, int & counter)
+{
+  for (const auto & act : story_node["Act"]) {
+    for (const auto & maneuver_group : act["ManeuverGroup"]) {
+      for (const auto & maneuver : maneuver_group["Maneuver"]) {
+        processManeuver(maneuver, counter);
+      }
+    }
+  }
+}
+
+void VisualizationConditionGroupsDisplay::processManeuver(
+  const YAML::Node & maneuver_node, int & counter)
+{
+  for (const auto & event : maneuver_node["Event"]) {
+    if (event["StartTrigger"] && event["StartTrigger"]["ConditionGroup"]) {
+      processEvent(event, counter);
+    }
+  }
+}
+
+void VisualizationConditionGroupsDisplay::processEvent(const YAML::Node & event_node, int & counter)
+{
+  std::string event_name;
+  try {
+    event_name = event_node["name"].as<std::string>();
+  } catch (const YAML::BadConversion & e) {
+    event_name = "";
+  }
+  if (event_name.empty()) {
+    event_name = "ConditionGroup" + std::to_string(counter++);
+  }
+
+  ConditionGroups condition_groups_msg;
```

Comment:
`ConditionGroups` is not a ros2/protobuf message.
I suggest to rename variable.

Reply:
changed in https://github.com/tier4/scenario_simulator_v2/pull/1033/commits/acdbc5a3017e305b20f8cf82013859e88642059f